src/main/java/Parser.java

```diff
@@ -0,0 +1,90 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a parser that will make sense of user commands and break down
+ * to various components for ease of processing.
+ */
+public class Parser {
+    /**
+     * Returns an array of string where the commands for tasks have been broken
+     * into its various parts for ease of processing/
+     * E.g. deadline task returns [deadline, task description, date, time]
+     * event = [event, task description, date, start time, end time]
+     * @param command The user command.
+     * @return An array of string which has been processed.
+     * @throws InvalidDateAndTimeException If input date/time is invalid.
+     * @throws InvalidTaskNumber If an improper integer task number is called in delete and done.
+     * @throws NoDescriptionException If there is no task description for addition of tasks.
+     * @throws NotTaskException If an invalid command is being input.
+     */
+    public String[] parse(String command) throws InvalidDateAndTimeException,
```

Comment:
Same with the previous switch statement, maybe you could separate out some of the code into separate methods to make this method shorter?